src/core/symbology/classification/qgsclassificationjenks.cpp

```diff
@@ -0,0 +1,171 @@
+/***************************************************************************
+    qgsclassificationjenks.h
+    ---------------------
+    begin                : September 2019
+    copyright            : (C) 2019 by Denis Rouzaud
+    email                : denis@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <limits> // for jenks classification
+#include "qgsclassificationjenks.h"
+
+QgsClassificationJenks::QgsClassificationJenks()
+  : QgsClassificationMethod( true /*valuesRequired*/,
+                             false /*symmetricMode*/ )
+{
+
+}
+
+QString QgsClassificationJenks::name() const
+{
+  return QObject::tr( "Natural Breaks (Jenks)" );
+}
+
+QString QgsClassificationJenks::id() const
+{
+  return QStringLiteral( "Jenks" );
+}
+
+QgsClassificationMethod *QgsClassificationJenks::clone() const
+{
+  QgsClassificationJenks *c = new QgsClassificationJenks();
+  copyBase( c );
+  return c;
+}
+
+
+QList<double> QgsClassificationJenks::calculateBreaks( double minimum, double maximum,
+    const QList<double> &values, int nclasses )
+{
+  // Jenks Optimal (Natural Breaks) algorithm
+  // Based on the Jenks algorithm from the 'classInt' package available for
+  // the R statistical prgramming language, and from Python code from here:
+  // http://danieljlewis.org/2010/06/07/jenks-natural-breaks-algorithm-in-python/
```

Comment:
note to self - we should resync this with classInt and double check we aren't missing any upstream fixes...